src/components/ExperienceForm/ExperienceForm.js

```diff
@@ -0,0 +1,153 @@
+          body: PropTypes.PropTypes.shape({
+            value: PropTypes.string
+          }),
+          field_experience_path: PropTypes.string.isRequired
+        })
+      )
+    })
+  };
+
+  static defaultProps = {
+    submitHandler: false,
+    experienceSlug: false,
+    experiences: {
+      error: null,
+      items: []
+    }
+  };
+
+  /**
+   * Fetches the experience that belongs to the given slug.
+   */
+  getExperience = () => {
+    const {
+      experienceSlug,
+      experiences: { items }
+    } = this.props;
+    return experienceSlug
+      ? items.find(item => item.field_experience_path === experienceSlug)
+      : false;
+  };
+
+  /**
+   * Handles dispatching create or update actions.
+   *
+   * @param {object} event - Submit event object.
+   */
+  handleSubmit = event => {
+    event.preventDefault();
+    const {
+      dispatch,
+      user,
+      history: { push },
+      experienceSlug
+    } = this.props;
+    dispatch({
+      type: experienceSlug ? EXPERIENCES_EDIT : EXPERIENCES_CREATE,
+      user,
+      id: this.getExperience().id,
+      title: event.target[0].value,
+      field_experience_path: event.target[1].value,
+      body: event.target[2].value,
+      successHandler: () => push('/dashboard')
+    });
+  };
+
+  /**
+   * {@inheretdoc}
+   */
+  render() {
+    const {
+      classes,
+      submitHandler,
+      experienceSlug,
+      experiences: { error }
+    } = this.props;
+
+    const experience = this.getExperience();
+
+    return (
+      <form onSubmit={submitHandler || this.handleSubmit}>
+        {error && <Message>{error}</Message>}
+        <TextField
+          id="title"
+          label="Title"
+          type="text"
+          required
+          helperText="Enter a user-friendly title for your experience."
+          defaultValue={experience ? experience.title : ''}
+          className={classes.textField}
+        />
+        <TextField
+          id="field_experience_path"
+          label="URL Path"
+          type="text"
+          required
+          helperText="Enter a name for your experience. For example, if you enter 'my-new-experience', your experience will be published to /experience/my-new-experience."
+          defaultValue={experience ? experience.field_experience_path : ''}
+          className={classes.textField}
+        />
+        <TextField
+          id="body"
+          label="Description"
+          type="text"
+          multiline
+          required
+          rows={6}
+          helperText="Describe your experience so people will know what to expect before they enter your experience."
+          defaultValue={
+            experience && experience.body ? experience.body.value : ''
+          }
+          className={classes.textField}
+        />
+        <Button
+          variant="raised"
+          color="primary"
+          type="submit"
+          className={classes.button}
+        >
+          {experienceSlug ? 'Save' : 'Create'}
```

Comment:
I feel like we could get away with this being being "Save" or "Save Experience" for both.

Reply:
That's a good point, I think "Save" would be fine :100: